plugins/woocommerce/client/legacy/js/frontend/a8c-address-autocomplete-service.js

```diff
@@ -0,0 +1,459 @@
+			var time = Date.now();
+			if ( shouldInvoke( time ) ) {
+				return trailingEdge( time );
+			}
+			// Restart the timer.
+			timerId = setTimeout( timerExpired, remainingWait( time ) );
+		}
+
+		function trailingEdge( time ) {
+			timerId = undefined;
+
+			// Only invoke if we have `lastArgs` which means `func` has been
+			// debounced at least once.
+			if ( trailing && lastArgs ) {
+				return invokeFunc( time );
+			}
+			lastArgs = lastThis = undefined;
+			return result;
+		}
+
+		function cancel() {
+			if ( timerId !== undefined ) {
+				clearTimeout( timerId );
+			}
+			// Reject any pending promise
+			if ( lastArgs && lastArgs._resolve ) {
+				lastArgs._resolve( [] ); // Resolve with empty array for cancelled requests
+			}
+			lastInvokeTime = 0;
+			lastArgs = lastCallTime = lastThis = timerId = undefined;
+		}
+
+		function flush() {
+			return timerId === undefined ? result : trailingEdge( Date.now() );
+		}
+
+		function debounced() {
+			var time = Date.now(),
+				isInvoking = shouldInvoke( time );
+
+			lastArgs = arguments;
+			lastThis = this;
+			lastCallTime = time;
+
+			if ( isInvoking ) {
+				if ( timerId === undefined ) {
+					return leadingEdge( lastCallTime );
+				}
+				if ( maxing ) {
+					// Handle invocations in a tight loop.
+					clearTimeout( timerId );
+					timerId = setTimeout( timerExpired, wait );
+					return invokeFunc( lastCallTime );
+				}
+			}
+			if ( timerId === undefined ) {
+				timerId = setTimeout( timerExpired, wait );
+			}
+			// Return a promise that will resolve when the function is eventually called
+			return new Promise( ( resolve ) => {
+				// Store the resolve function to be called when the function executes
+				lastArgs._resolve = resolve;
+			} );
+		}
+		debounced.cancel = cancel;
+		debounced.flush = flush;
+		return debounced;
+	}
+
+	Object.entries( a8cAddressAutocompleteServiceKeys ).forEach(
+		( [ key, value ] ) => {
+			let sessionId =
+				crypto && crypto.randomUUID
+					? crypto.randomUUID()
+					: Math.random().toString( 36 ).substring( 2 );
+			let requestDurations = [];
+			let serviceErrorRetries = 0;
+			// Cache for search results - key: `${inputValue}:${country}`, value: data
+			const searchCache = new Map();
+
+			// Helper function to check cache
+			const getCachedResult = ( inputValue, country ) => {
+				const cacheKey = `${ inputValue }:${ country }`;
+				return searchCache.get( cacheKey ) || null;
+			};
+
+			// Helper function to store result in cache
+			const cacheResult = ( inputValue, country, data ) => {
+				const cacheKey = `${ inputValue }:${ country }`;
+				searchCache.set( cacheKey, data );
+
+				// Clean up cache if it gets too large (more than 100 entries)
+				if ( searchCache.size > 100 ) {
+					// Remove oldest entries (first 20 entries)
+					const entries = Array.from( searchCache.entries() );
+					entries.slice( 0, 20 ).forEach( ( [ key ] ) => {
+						searchCache.delete( key );
+					} );
```

Comment:
```suggestion
					const keys = Array.from( searchCache.keys() );
					keys.slice( 0, 20 ).forEach( ( [ key ] ) => {
						searchCache.delete( key );
					} );
```
Looks like you only use keys here.